common/persistence/nexus_service_manager.go

```diff
@@ -48,16 +52,20 @@ var (
 type (
 	nexusServiceManagerImpl struct {
 		persistence NexusServiceStore
+		serializer  serialization.Serializer
+		logger      log.Logger
 	}
 )
 
-var _ NexusServiceManager = (*nexusServiceManagerImpl)(nil)
-
 func NewNexusServiceManager(
 	persistence NexusServiceStore,
+	serializer serialization.Serializer,
+	logger log.Logger,
 ) NexusServiceManager {
```

Comment:
The recommendation in Go is to return structs and accept interfaces.
This whole interface seems redundant to me, why do we need it?

Reply:
What about it is redundant? I believe we need the separate interface definition so that our client wrappers will work. For example:
https://github.com/temporalio/temporal/blob/main/common/persistence/persistence_metric_clients.go
https://github.com/temporalio/temporal/blob/main/common/persistence/client/factory.go#L112

Reply:
Oh I see, that makes sense to me. If this is wrapped by our generated clients, we need the interface.